lib/foreman_ansible_core/runner/ansible_runner.rb

```diff
@@ -109,15 +111,28 @@ def write_playbook
       def start_ansible_runner
         env = {}
         env['FOREMAN_CALLBACK_DISABLE'] = '1' if @rex_command
-        command = [env, 'ansible-runner', 'run', @root, '-p', 'playbook.yml']
-        command << '--cmdline' << '"--check"' if check_mode?
-        command << verbosity if verbose?
-        initialize_command(*command)
-        logger.debug("[foreman_ansible] - Running command '#{command.join(' ')}'")
+        command = "ansible-runner run #{@root} -p playbook.yml #{cmdline} #{verbosity}"
+        initialize_command(env, command)
```

Comment:
Is there a reason why we switch from the array form to an escaped string? It feels less safe, especially if part of the string is unsanitized user input

Reply:
The quotes were not correctly recognized when using array which was causing malformed commands.

Reply:
Which quotes?

Reply:
Would it work as array if we would pass all `cmdline` as one arg?
```ruby
command << '--cmdline' << cmdline
```

Reply:
> Which quotes?
Quotes delimiting an argument for runner command line: `--cmdline "--tags foo"`

Reply:
> Would it work as array if we would pass all `cmdline` as one arg?
> 
> ```ruby
> command << '--cmdline' << cmdline
> ```

Nope, that does not work either.

Reply:
> Quotes delimiting an argument for runner command line: `--cmdline "--tags foo"`

Represented in array form as `['--cmdline', '"--tags foo"']`?

Reply:
> > Quotes delimiting an argument for runner command line: `--cmdline "--tags foo"`
> 
> Represented in array form as `['--cmdline', '"--tags foo"']` or `['--cmdline "--tags foo"']`?

I tried both, neither works as expected.

Reply:
Well, neither of those is supposed to work as there are superfluous quotes. `['--cmdline=foo,baz']` or `['--cmdline', 'foo,baz']` should work

Reply:
we need those quotes tho, it is one argument, that contains arguments for some subprocess

Reply:
one way around it, would be dumping these args to folder, which ansible-runner reads. I think it is `cmd`

Reply:
Do you mean [env/cmdline](https://ansible-runner.readthedocs.io/en/stable/intro.html#env-cmdline) or is there something else?

Reply:
> we need those quotes tho, it is one argument, that contains arguments for some subprocess

The array form is not processed by a shell, a single entry in the array is a single argument being passed to the child process, I wouldn't say the quotes are necessary.

> ['--cmdline', 'foo,baz'] gives me --cmdline foo,baz while I need --cmdline "--tags foo,baz". The --tags flag together with list of the tags is an arg for the --cmdline.

Right, my bad. I meant `['--cmdline', '--tags=foo,baz']`

Reply:
I haven't tried with the `=` which might help just for tags but when I add a flag for check mode into the mix, I still need those quotes: `--cmdline "--tags=foo,baz --check"`